zipline/algorithm.py

```diff
@@ -266,6 +266,7 @@ def __init__(self, *args, **kwargs):
         self.algoscript = kwargs.pop('script', None)
 
         self._initialize = None
+        self._analyze = kwargs.pop('analyze', None)
```

Comment:
From the two cases below, it looks like we should assign to `self._analyze` from `kwargs` only in the second case, where we assign all our functions from `kwargs`.  (In the first case, we assign it from `self.namespace`.)  What do you think about moving this assignment to the body of the `elif` below?  

Also, did you find this worked for you, since we re-assign `self._analyze = None` two lines below this?